dev/SearchBox.js

```diff
@@ -1,6 +1,9 @@
 import React, {Component, PropTypes as T} from 'react';
 import connect from '../src/connect.js';
 
-export default connect()(function({helper}) {
-  return <input onChange={(e) => helper.setQuery(e.target.value).search()} />;
+export default connect()(function({helper, search}) {
```

Comment:
I believe this behavior could be the default, yes. However, it might break existing code in some edge cases where we rely on it to be synchronous. I can't think of a case where batching wouldn't make sense.

[I ended up monkey-patching search anyway in my last experiment](https://github.com/algolia/instantsearch.js/commit/7b462a0caa9e0797dca845ac4f4a1be8d9636e01#diff-af5c8b1cd68eb636d071dfc6abe81b28R38) in order to add some default state to the helper while keeping its normal state untouched. This also means that I don't need a debounced search anymore.

So the API I need might actually be a way to register/unregister hooks on `search` in order to manipulate the parameters of a search.